calc/src/mechanics/util.ts

```diff
@@ -356,6 +357,34 @@ export function getFinalDamage(
   return OF16(pokeRound(Math.max(1, OF32(damageAmount * finalMod) / 0x1000)));
 }
 
+/**
+ * Determines which move category Shell Side Arm should behave as. Applications
+ * that consume this library may want to import this function to show users move
+ * category information.
+ *
+ * Notes:
+ * 1. As far as I can tell, research hasn't been tone on whether it's the larger
+ * ratio (atk / def) or the difference (atk - def) which decides the move
+ * category.
+ * Ratio seems more likely to me, since the pokemon damage formula is
+ * dependent on the ratio and not the difference of attack and defense stats.
+ *
+ * 2. in game, when stats are equal, category is random.  For damage, it won't
+ * matter most of the time (items like assault vest aren't included in this
+ * decision though, so *may* take effect).
+ *
+ * {@link https://www.reddit.com/r/VGC/comments/hcfe4k/galarian_slowbro_shell_side_arm_breakdown/}
+ *
+ * @param source Attacking pokemon (after stat modifications)
+ * @param target Target pokemon (after stat modifications)
+ * @returns 'Physical' | 'Special'
+ */
+export function getShellSideArmCategory(source: Pokemon, target: Pokemon): MoveCategory  {
+  const physicalDamage = source.stats.atk / target.stats.def;
+  const specialDamage = source.stats.spa / target.stats.spd;
+  return physicalDamage > specialDamage ? 'Physical' : 'Special'
```

Comment:
This is not the correct formula.
```suggestion
  const physicalDamage = ((((2 * source.level / 5 + 2) * 90 * source.stats.atk) / target.stats.def) / 50);
  const specialDamage = ((((2 * source.level / 5 + 2) * 90 * source.stats.spa) / target.stats.spd) / 50);
  return physicalDamage > specialDamage ||
    (physicalDamage === specialDamage && Math.random() < 0.5)
    ? 'Physical'
    : 'Special';
```

(I'm unsure if this is the right way to do a random choice to simulate a coinflip)

Reply:
Damage cannot be stochastic. You have two choices here:

- either simply favor one over the other and leave a comment explaining this decision = the approach ajhyndman originally went with
- add a field to `Move` to configure how the tie should be broken (so that it could be wired up to a UI which allows the user to decide) = more correct, but potentially overkill

Reply:
>  const physicalDamage = ((((2 * source.level / 5 + 2) * 90 * source.stats.atk) / target.stats.def) / 50);
>  const specialDamage = ((((2 * source.level / 5 + 2) * 90 * source.stats.spa) / target.stats.spd) / 50);

I recognise that this is technically correct, but since every other value is the same in both formulas, the relative ratio of attack / def is sufficient to determine which will be more damage.

To do the long form:

```
// physical
= ((((2 * source.level / 5 + 2) * 90 * source.stats.atk) / target.stats.def) / 50);
= ((((2 * 50 / 5 + 2) * 90 * source.stats.atk) / target.stats.def) / 50);
= ((((100 / 5 + 2) * 90 * source.stats.atk) / target.stats.def) / 50);
= ((((20 + 2) * 90 * source.stats.atk) / target.stats.def) / 50);
= (((22 * 90 * source.stats.atk) / target.stats.def) / 50);
= (((22 * 90 * source.stats.atk) / target.stats.def) / 50);
= (((1980 * source.stats.atk) / target.stats.def) / 50);
= ((1980 * (source.stats.atk / target.stats.def) / 50)
= ((1980 / 50) * (source.stats.atk / target.stats.def))
= (39.6 * (source.stats.atk / target.stats.def))

// special
= ((((2 * source.level / 5 + 2) * 90 * source.stats.spa) / target.stats.spd) / 50);
// ...
= (39.6 * (source.stats.spa / target.stats.spd))

// now do the comparison
(39.6 * (source.stats.atk / target.stats.def)) > (39.6 * (source.stats.spa / target.stats.spd))
// divide both sides by a postive real number (no need to reverse the direction of the inequality)
(source.stats.atk / target.stats.def)) > (source.stats.spa / target.stats.spd)
```

I noticed that the showdown implementation does the discrete Math.floor at each stage of computation, which I suppose _could_ matter for some close calls.  But I didn't see any research on those specific cases to verify that is in fact important.

Reply:
I don't expect anyone cares about this much detail, but I think I can prove mathematically that the discrete flooring will never change which type wins as well.

Referencing the showdown implementation:

```javascript
Math.floor(Math.floor(Math.floor(Math.floor(2 * pokemon.level / 5 + 2) * 90 * atk) / def) / 50);
```

The expression `Math.floor(2 * pokemon.level / 5 + 2)` will always be an integer after `Math.floor`.  If we simplify this to `C`:

```javascript
Math.floor(Math.floor(Math.floor(C * 90 * atk) / def) / 50);
```

C, 90 and atk are all integer values, so multiplying them will also always result in an integer.  That means the `Math.floor` call wrapping this is redundant, and can be dropped.

```javascript
Math.floor(Math.floor((C * 90 * atk) / def) / 50);
```

Multiplication is commutative, so you can drop those wrapping parens and see that the value of atk / def is computed before truncation, and so the resulting damage type depends on the exact ratio of atk / def.

---

With all that said, _IF_ the Smogon assumptions are accurate, the only remaining issue I can imagine is that sometimes the game might flatten a few close cases and randomly choose a type.  So we won't be _wrong_, but in some super rare case, we might make a different fixed guess than usual.